### PR TITLE
Fix fetching shipping method

### PR DESCRIPTION
**Pull request by kevinxh**
The validation of `inputAddres` was broken. This PR fixes the problem.

 **JIRA**: (link to JIRA ticket)
 **Linked PRs**: (links to corresponding PRs, optional)

## Changes
- fix `hasShippingMethods` selector
- move shipping address validation function `shouldFetchShippingMethods` to hybris connector

## Todos
- [ ] (if applicable) analytics events instrumented to measure impact of change
- [ ] Add a high-level description of your changes to the CHANGELOG.md, including a link to the PR with your changes

## How to test-drive this PR
- (necessary config changes)
- Run `npm start`
- Preview [Merlin's Potions](https://preview.mobify.com/?url=https%3A%2F%2Fwww.merlinspotions.com%2F&site_folder=https%3A%2F%2Flocalhost%3A8443%2Floader.js&disabled=0&domain=&scope=1)
- (how to access the new / changed functionality -- fixtures, URLs)
#

### CODE REVIEW

File: web/app/connectors/hybris-connector/checkout/commands.js [hunk 1]
```diff
@@ -129,9 +129,31 @@ const shouldUpdateShippingAddress = (inputAddress) => (dispatch, getState) => {
...
+}
+
 export const fetchShippingMethodsEstimate = (inputAddress) => (dispatch, getState) => {
+    if (!inputAddress || !shouldFetchShippingMethods(inputAddress)) {
+        return Promise.resolve()
+    }
+    
     const shippingMethods = getShippingMethods(getState()).toJS()
-    const avoidUpdating = !dispatch(shouldUpdateShippingAddress(inputAddress)) && shippingMethods.length
+    const avoidUpdating = !shouldUpdateShippingAddress(inputAddress) && shippingMethods.length
```
Comment (juanjcampos) [inline web/app/connectors/hybris-connector/checkout/commands.js]: Why did you remove the dispatch(...) from here? Notice that `shouldUpdateShippingAddress(inputAddress)` does not return a value, but a function:
`const shouldUpdateShippingAddress = (inputAddress) => (dispatch, getState) => {`

Comment (kevinxh) [reply web/app/connectors/hybris-connector/checkout/commands.js]: Mistakes were made... fixed :)

File: web/app/containers/checkout-shipping/partials/shipping-address.jsx [hunk 2]
```diff
@@ -18,146 +18,123 @@ import FieldRow from 'progressive-web-sdk/dist/components/field-row'
 import ShippingAddressFields from './shipping-address-fields'
 import {UI_NAME} from 'progressive-web-sdk/dist/analytics/data-objects/'
 
-class ShippingAddressForm extends React.Component {
-    componentWillReceiveProps(nextProps) {
```
Comment (kevinxh) [inline web/app/containers/checkout-shipping/partials/shipping-address.jsx]: Delete this lifecycle function and change `ShippingAddressForm` to a stateless component, since https://github.com/mobify/platform-scaffold/pull/1022 solves the shipping method initialization after user signs in.
